### PR TITLE
Add Doc for layouts, string builder, int long, and string util

### PR DESCRIPTION
**Pull request by yicheng4**

### CODE REVIEW

**Comment by yicheng4** [comment]
Not sure whether to make the new folder util or lib

File: doc/tutorial/Layouts.md [hunk 1]
```diff
@@ -74,6 +74,25 @@ var last = str.name[str.length - 1];     // access is dynamically bounds-checked
 Repeated fields allow expressing fixed-size arrays inside a layout.
 Accesses to repeated fields are indexed with `[]` and the index is bounds-checked against the statically-declared number of elements in the field.
 
+## Field Overlapping and Memory Skipping in Layouts
+In Virgil layouts, field overlapping within a layout is strictly prohibited to prevent data corruption and ensure data integrity. Each field must have a unique, non-overlapping byte offset. 
+
+```
+layout DeviceRegister {
+       +0    command:   u16;    
+       +1    status:    u16;    // Overlaps, illeagal
```
Comment (titzer) [inline doc/tutorial/Layouts.md:83]: spelling: illegal

File: doc/tutorial/util/StringBuilder.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- `.putc(char)` - Appends a single character.
+- `.putd(decimal)`, `.putx(hexadecimal)`, and `.putz(boolean)` - Used for appending numeric and boolean data in various formats.
+
+### Output and Reset
+- `.toString()` - Returns the current string value of the buffer and deletes the StringBuilder instance
+- `.reset()` - Clears the buffer for reuse.
+- `.grow(size)` - Suggests a new capacity for the buffer to optimize performance.
+
+### Advanced Data Handling
+- `.putk("substring", start, end)` - Adds a substring based on specified range.
```
Comment (titzer) [inline doc/tutorial/util/StringBuilder.md:27]: It's worth mentioning `putr` here too, as that's generally preferrable now to doing and explicit start and end.

File: doc/tutorial/util/StringBuilder.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+- `.pad(char, length)` - Pads the buffer with a character up to a specified length.
+
+### Utility Functions
+- `.ln()` - Appends a newline.
+- `.tab()` - Adds a tabulation character.
```
Comment (titzer) [inline doc/tutorial/util/StringBuilder.md:32]: Just "tab".

**Review by titzer — COMMENTED** [review]
Looks good to me with minor comments. Thanks for doing this!

**Pull request merged**